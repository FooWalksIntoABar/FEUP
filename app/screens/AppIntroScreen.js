import React, { Component } from 'react'
import AppIntroSlider from 'react-native-app-intro-slider'
import PropTypes from 'prop-types'
import Images from '../assets/images/images'

const slides = [
  {
    key: '1',
    title: 'Texto explicativo #1',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#59b2ab'
  },
  {
    key: '2',
    title: 'Texto explicativo #2',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#febe29'
  },
  {
    key: '3',
    title: 'Texto explicativo #3',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#22bcb5'
  }
]

export default class AppIntroScreen extends Component {
  _onDone = () => {
    this.props.navigation.goBack()
  }

  render () {
    return (
      <AppIntroSlider slides={slides} onDone={this._onDone} />
    )
  }
}

AppIntroScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
