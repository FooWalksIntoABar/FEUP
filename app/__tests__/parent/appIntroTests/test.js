import 'react-native'
import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import AppIntroScreen from '../../../screens/parent/AppIntroScreen'

configure({ adapter: new Adapter() })

describe('AppIntroScreen snapshot', () => {
  it('renders AppIntroScreen correctly', () => {
    const wrapper = shallow(<AppIntroScreen
      navigation={{ navigate: jest.fn() }} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })
})
