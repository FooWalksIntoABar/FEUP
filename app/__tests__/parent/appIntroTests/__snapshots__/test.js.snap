// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIntroScreen snapshot renders AppIntroScreen correctly 1`] = `
<AppIntroSlider
  activeDotStyle={
    Object {
      "backgroundColor": "rgba(255, 255, 255, .9)",
    }
  }
  buttonStyle={null}
  buttonTextStyle={null}
  doneLabel="Done"
  dotStyle={
    Object {
      "backgroundColor": "rgba(0, 0, 0, .2)",
    }
  }
  nextLabel="Next"
  onDone={[Function]}
  prevLabel="Back"
  skipLabel="Skip"
  slides={
    Array [
      Object {
        "backgroundColor": "#59b2ab",
        "image": 1,
        "imageStyle": Object {
          "height": 240,
          "width": 240,
        },
        "key": "1",
        "text": "A ser preenchido pelas nossas fantásticas product owners.
O número de páginas e imagem é ajustável consoante necessário.",
        "title": "Texto explicativo #1",
      },
      Object {
        "backgroundColor": "#febe29",
        "image": 1,
        "imageStyle": Object {
          "height": 240,
          "width": 240,
        },
        "key": "2",
        "text": "A ser preenchido pelas nossas fantásticas product owners.
O número de páginas e imagem é ajustável consoante necessário.",
        "title": "Texto explicativo #2",
      },
      Object {
        "backgroundColor": "#22bcb5",
        "image": 1,
        "imageStyle": Object {
          "height": 240,
          "width": 240,
        },
        "key": "3",
        "text": "A ser preenchido pelas nossas fantásticas product owners.
O número de páginas e imagem é ajustável consoante necessário.",
        "title": "Texto explicativo #3",
      },
    ]
  }
/>
`;
